integration-tests/src/test/java/oracle/weblogic/kubernetes/ItExternalRmiTunneling.java

```diff
@@ -0,0 +1,502 @@
+    deployIngress.append(Paths.get(RESULTS_ROOT, "voyager.tunneling.yaml"));
+    assertDoesNotThrow(() -> exec(new String(deployIngress), true));
+
+    StringBuffer deployTlsIngress = new StringBuffer("kubectl delete -f ");
+    deployTlsIngress.append(Paths.get(RESULTS_ROOT, "voyager.tls.tunneling.yaml"));
+    assertDoesNotThrow(() -> exec(new String(deployTlsIngress), true));
+
+    // uninstall Voyager
+    if (voyagerHelmParams != null) {
+      assertThat(uninstallVoyager(voyagerHelmParams))
+          .as("Test uninstallVoyager returns true")
+          .withFailMessage("uninstallVoyager() did not return true")
+          .isTrue();
+    }
+  }
+
+  // Create and display SSL certificate and key using openssl with SAN extension
+  private static void createCertKeyFiles(String cn) {
+
+    Map<String, String> sanConfigTemplateMap  = new HashMap();
+    sanConfigTemplateMap.put("INGRESS_HOST", K8S_NODEPORT_HOST);
+
+    Path srcFile = Paths.get(RESOURCE_DIR, 
+        "tunneling", "san.config.template.txt");
+    Path targetFile = assertDoesNotThrow(
+        () -> generateFileFromTemplate(srcFile.toString(), 
+        "san.config.txt", sanConfigTemplateMap));
+    logger.info("Generated SAN config file {0}", targetFile);
+
+    assertDoesNotThrow(() -> {
+      tlsKeyFile = Paths.get(RESULTS_ROOT, domainNamespace + "-tls.key");
+      tlsCertFile = Paths.get(RESULTS_ROOT, domainNamespace + "-tls.cert");
+      String command = "openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout " + tlsKeyFile
+          + " -out " + tlsCertFile + " -subj \"/CN=" + cn + "\" -extensions san"
+          + " -config " + Paths.get(RESULTS_ROOT, "san.config.txt");
+      logger.info("Executing command: {0}", command);
+      ExecResult result = ExecCommand.exec(command, true);
+      logger.info("openssl command  returned {0}", result.toString());
+    });
+
+    assertDoesNotThrow(() -> {
+      String command2 = "openssl x509 -in " + tlsCertFile + " -noout -text ";
+      logger.info("Executing command: {0}", command2);
+      ExecResult result = ExecCommand.exec(command2, true);
+      logger.info("openssl list command  returned {0}", result.toString());
+    });
+  }
+
+  // Import the certificate into a JKS TrustStore to be used while running 
+  // external JMS client to send message to WebLogic.
+  private void createJksStore() {
+    assertDoesNotThrow(() -> {
+      jksTrustFile = Paths.get(RESULTS_ROOT, domainNamespace + "-trust.jks");
+      String command = "keytool -import -file " + tlsCertFile 
+             + " --keystore " + jksTrustFile 
+             + " -storetype jks -storepass password -noprompt ";
+      logger.info("Executing command: {0}", command);
+      ExecCommand.exec(command, true);
+    });
+
+    assertDoesNotThrow(() -> {
+      String command2 = "keytool -list -keystore " + jksTrustFile + " -storepass password -noprompt";
+      logger.info("Executing command: {0}", command2);
+      ExecCommand.exec(command2, true);
```

Comment:
you don't have to check the result?

Reply:
Modified the logic to check the ResultSet's exitcode

Reply:
Modified logic to use actions.impl.primitive.Command to execute bash command.